### ISSUE TITLE
Support for additional sprig template functions

### ISSUE THREAD

**Issue by nothingofuse** [crossplane-contrib/function-go-templating#102]
### What problem are you facing?
I want to use function-go-templating to do things like:
* write conditionals based on strings containing substrings
* split strings to reconstruct pieces of a string into other resource fields
* find unique items in a list to create resources based only on those unique items
* trim suffixes or prefixes from strings

### How could this Function help solve your problem?

By implementing more of the functions from the masterminds/sprig library the function is using to enable support for more of the library's functions, primarily the functions in 
http://masterminds.github.io/sprig/defaults.html
http://masterminds.github.io/sprig/strings.html
http://masterminds.github.io/sprig/lists.html

**Comment by LCaparelli** [comment]
I would also add that supporting Sprig functions would assist engineers that are already used to Helm, which provides the same Spring functions. Familiarity with Helm seems to be one of the goals of this function, or at least being as familiar to the user as possible, a **very** nice-to-have feature in that sense.

**Comment by jaylevin** [comment]
This function already supports all the Sprig functions.


From the README:
![Uploading IMG_9881.jpeg…]()

**Comment by LCaparelli** [comment]
@jaylevin You're right, the function that I was looking for and couldn't find was `truncate`, which doesn't exist and I really meant to use `trunc` :sweat_smile: .

Thanks, sorry for the noise.
